File: solutions/fsharp/high-scores/HighScores.fs
module HighScores

let scores (values: int list): int list = values

let latest (values: int list): int = List.last values

let personalBest (values: int list): int = List.max values

let personalTopThree (values: int list): int list = List.sortDescending values |> List.truncate 3

File: solutions/scala/high-scores/src/main/scala/HighScores.scala
object HighScores:
  def latest(scores: List[Int]): Int = scores.last

  def personalBest(scores: List[Int]): Int = scores.max

  def personalTop(scores: List[Int]): List[Int] =
    scores.sorted.reverse.take(3)

  def report(scores: List[Int]): String =
    val last = latest(scores)
    val best = personalBest(scores)
    val diff = if last == best then ""
    else s" ${best - last} short of"

    s"Your latest score was $last. That's$diff your personal best!"
